x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/LookupJoin.java

```diff
@@ -71,4 +76,40 @@ protected NodeInfo<Join> info() {
             config().rightFields()
         );
     }
+
+    @Override
+    public void postAnalysisVerification(Failures failures) {
+        super.postAnalysisVerification(failures);
+        right().forEachDown(EsRelation.class, esr -> {
+
+            if (esr.indexMode().equals(IndexMode.LOOKUP)) {
```

Comment:
This `if` has become redundant. The right hand side of the `LOOKUP JOIN` _must_ have `IndexMode.LOOKUP`, otherwise something is wrong.

I'd remove this `if`, otherwise we might mask failures. In fact, given that we rely on `esr.indexMode` in the optimizer (maybe questionably, so?), I think we should either
- add an assertion that `esr.indexMode` is `IndexMode.LOOKUP`, or
- add an outright check that fails with `InvalidStateException`.